src/Lib.P2P/PubSub/FloodRouter.cs

```diff
@@ -0,0 +1,284 @@
+            {
+                log.Warn("Join topic failed.", e);
+            }
+        }
+
+        /// <inheritdoc />
+        public async Task LeaveTopicAsync(string topic, CancellationToken cancel)
+        {
+            localTopics.TryRemove(topic, out _);
+            var msg = new PubSubMessage
+            {
+                Subscriptions = new Subscription[]
+                {
+                    new Subscription
+                    {
+                        Topic = topic,
+                        Subscribe = false
+                    }
+                }
+            };
+            try
+            {
+                var peers = Swarm.KnownPeers.Where(p => p.ConnectedAddress != null);
+                await SendAsync(msg, peers, cancel).ConfigureAwait(false);
+            }
+            catch (Exception e)
+            {
+                log.Warn("Leave topic failed.", e);
+            }
+        }
+
+        /// <inheritdoc />
+        public Task PublishAsync(PublishedMessage message, CancellationToken cancel)
+        {
+            if (tracker.RecentlySeen(message.MessageId))
+                return Task.CompletedTask;
+
+            // Find a set of peers that are interested in the topic(s).
+            // Exclude author and sender
+            var peers = message.Topics
+               .SelectMany(topic => RemoteTopics.GetPeers(topic))
+               .Where(peer => peer != message.Sender)
+               .Where(peer => peer != message.Forwarder);
+
+            // Forward the message.
+            var forward = new PubSubMessage
+            {
+                PublishedMessages = new PublishedMessage[] {message}
+            };
+
+            return SendAsync(forward, peers, cancel);
+        }
+
+        private Task SendAsync(PubSubMessage msg, IEnumerable<Peer> peers, CancellationToken cancel)
+        {
+            // Get binary representation
+            byte[] bin;
+            using (var ms = new MemoryStream())
+            {
+                Serializer.SerializeWithLengthPrefix(ms, msg, PrefixStyle.Base128);
+                bin = ms.ToArray();
+            }
+
+            return Task.WhenAll(peers.Select(p => SendAsync(bin, p, cancel)));
+        }
+
+        private async Task SendAsync(byte[] message, Peer peer, CancellationToken cancel)
+        {
+            try
+            {
+                using (var stream = await Swarm.DialAsync(peer, ToString(), cancel).ConfigureAwait(false))
+                {
+                    await stream.WriteAsync(message, 0, message.Length, cancel).ConfigureAwait(false);
+                    await stream.FlushAsync(cancel).ConfigureAwait(false);
+                }
+
+                log.Debug($"sending message to {peer}");
+                return;
+            }
+            catch (Exception e)
+            {
+                log.Debug($"{peer} refused pubsub message.", e);
+            }
+        }
+
+#pragma warning disable VSTHRD100 // Avoid async void methods
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Do not suppress issues.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4612924)